### PR TITLE
Error if there is not a valid configuration file

### PR DESCRIPTION
**Pull request by GordonDiggs**
If there is no eslintrc.* file, ESLint's behavior is to do nothing (and
therefore emit no issues). Because this can be confusing, we should fail
hard and show helpful information.

screen shot 2016-02-12 at 10 54 48 am

@codeclimate/review

### CODE REVIEW

File: bin/eslint.js [hunk 1]
```diff
@@ -232,4 +231,12 @@ function analyzeFiles() {
...
 }
 
-analyzeFiles();
+if (validateConfig(options.configFile)) {
+  console.error("ESLint is running with the " + cli.getConfigForFile(null).parser + " parser.");
+
+  analyzeFiles();
+} else {
+  console.error("No rules are configured. Make sure you have added a config file with rules enabled.");
+  console.error("See our documentation at docs.codeclimate.com/docs/eslint for more information.");
```
Comment (wfleming) [inline bin/eslint.js]: This is a nit, but can we put a protocol on the URL?

Comment (ABaldwinHunter) [inline bin/eslint.js]: What do you think about specifying that it's a `.eslintrc` or `.eslintrc.js` config file?

File: test/validate_config_test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      if (err) throw err;
+
+      process.chdir(directory);
+
+      var configPath = path.join(directory, ".eslintrc.json");
+      var config = {
+        rules: {
+          strict: 0
+        }
+      };
```
Comment (ABaldwinHunter) [inline test/validate_config_test.js]: nice tests!

File: bin/eslint.js (hunk 1 above)
Comment (GordonDiggs) [inline bin/eslint.js]: @wfleming yep sounds good

@ABaldwinHunter I think there are too many possibilities to list here (`.eslintrc(.js|.json|.yml.yaml|nothing`) and it would be a bit wordy. My link to the docs was so that people would see it there

Comment (ABaldwinHunter) [inline bin/eslint.js]: Okay sounds good :+1:

**Comment by ABaldwinHunter** [comment]
@GordonDiggs one optional comment but LGTM!
love the specs.

Unrelated to shipping this change, but wdyt about adding having this validation tripped in `codeclimate validate-config`?

**Comment by ABaldwinHunter** [comment]
Might be outside the purview of validating the `.codeclimate.yml`.

**Comment by wfleming** [comment]
> Unrelated to shipping this change, but wdyt about adding having this validation tripped in codeclimate validate-config?

Personally I think the CLI already knows too much about specific engines. This is an eslint behavior, not a general engine behavior, so it doesn't seem appropriate to me for the CLI to do this.

**Comment by GordonDiggs** [comment]
I agree with @wfleming. Additionally, in the ideal workflow, a user will call `codeclimate init` first and never see this.

File: lib/validate_config.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+var CLIEngine = require("eslint").CLIEngine
+  , cli = new CLIEngine()
+  , fs = require("fs");
+
+module.exports = function(configPath) {
+  if (configPath) {
```
Comment (wfleming) [inline lib/validate_config.js]: This behavior is a little surprising to me: this would mean if you explicitly told us to use `.eslintrc.foo` as a config file, but that file didn't exist, we would still pass that as valid?

Comment (GordonDiggs) [inline lib/validate_config.js]: @wfleming yeah - if you've specified a specific config file this check would pass, but a later one would error.

Comment (wfleming) [inline lib/validate_config.js]: Gotcha. Makes sense.
